### ISSUE TITLE
IE 11 Guidance - Invariant Violation with core-js polyfill

### ISSUE THREAD

**Issue by c-eiser13** [intljusticemission/react-big-calendar#1284]
#### What's the current behavior?
First off, this is a great component and works as expected in Chrome and FF, however I am forced to support IE 11. I am working on a SharePoint Framework web part that has a React component for displaying RBC. When I first viewed the page in IE, the component was missing and upon inspecting dev tools, I get the below error:

![error1](https://user-images.githubusercontent.com/42096464/56039792-b7fe8180-5cfa-11e9-8fec-a980c68b9315.PNG)

From there, I imported the core-js/es6/object polyfill at the top of my imports for this web part:

![imports](https://user-images.githubusercontent.com/42096464/56039911-ebd9a700-5cfa-11e9-89df-0dc94618f726.PNG)

Now when I refresh my page in IE, the calendar renders for about 3-5 seconds, then the component disappears. Upon inspecting the console, I get an Invariant Violation with a link to the error, which is: Objects are not valid as a React child (found: object with keys {$$typeof, type, key, ref, props, _owner}).

![violation](https://user-images.githubusercontent.com/42096464/56040163-699db280-5cfb-11e9-9ee4-d61e46576b39.PNG)

Solution Info:
"react": "16.7.0"
"react-dom": "16.7.0"
"react-big-calendar": "^0.20.4"

I'm experiencing this issue both while debugging and in a production build. I've tried downgrading RBC versions, but get the same issue with the core-js polyfills included, the calendar loads for 3-5 seconds, then the app crashes with the above error message. I've also tried the react-app-polyfill, but the calendar does not load at all when using this one.

If there is specific guidance somewhere on IE support, I'll gladly take a look, but was unable to find any. If further info is needed, I can provide as well.

#### What's the expected behavior?
Functional calendar in IE 11.

**Comment by PzYon** [comment]
I am having the same problem, also using SharePoint Framework and IE 11.

@jquense I understand you support IE10+ according the readme. I tried to build the examples so I could have a look at the above problem. However that's not possible as e.g. `Object.entries` (used in `App.js`) is not supported in IE. And even when I replace that using `Object.keys`, I get the next error (`Invalid character` - pointing to a `). Most probably that's also not supported in IE. Or do I need to be build the examples differently?

**Comment by jquense** [comment]
ya'll have to provide the necessary polyfills your app needs to run in IE10+, If we are inadvertantly relying on polyfills that aren't documented please let us know where or even PR a change to remove the builtin method

**Comment by PzYon** [comment]
@c-eiser13 , I managed to solve this problem by adding a polyfill for `Object.assign`. The important thing is that it internally doesn't use `objectInstance.hasOwnProperty` but instead `Object.prototype.hasOwnProperty.call`. 

I took the example from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign), added it is early as possible and didn't check if it is already defined or not (i.e. the usual polyfill pattern), so something like follows:

```
if (isIE) {
    Object.assign = function assign(target, varArgs) {
        // polyfill from MDN
    }
}
```

The problem isn't really related to react big calendar but to a polyfill for `Object.assign` that I believe is loaded by SPFx and that internally doesn't use the `hasOwnProperty`-function from the prototype but instead from the instance. For some reason I think certain objects created in IE don't have all members from the prototype or something. Don't really know the details, but that's what it seemed like for me...

**Comment by c-eiser13** [comment]
Thanks @PzYon just circling back to this issue today. Are you adding the polyfill in one of the lifecycle hooks of the React component, or in the webpart.ts file? I have tried it in componentDidMount, but it is still crashing in IE. Appreciate any info you may have!

**Comment by PzYon** [comment]
We have a polyfills file containing all the polyfills we need. As I wrote above, if I remember correctly, it's important to 
- take the [example](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) from MDN 
- and always override any existing polyfill/function in IE, as otherwise it won't work

Have you tried adding it as early as possible, e.g. in the `onInit` of your implementation of the `BaseClientSideWebPart`-class provided by SPFx? I hope it works for you too. As I wrote above I didn't really understand the problem 100%, just found that that fixes it.
